### PR TITLE
docs: add newer version alert messages

### LINKED ISSUE #1575: Add a toast in our documentation for new releases
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

Study if this should be a toast or an alert. Add a toast in our documentation to inform people that a new release has been done.

The toast should:
- contain a link to redirect to the migration guide
- contain a close button
- contain the version number
- be displayed only 1-2 week after the release (or for the first visit but would need cookies that we don't have yet and have some troubles with GDPR)
- be displayed until it's closed
- be displayed if possible in previous versions and in the whole documentation

Feel free to propose something.

### Motivation and context

Help users to know when a new release has been released.

### PR DESCRIPTION
**Pull request by julien-deramond**
⚠️ Must be merged once v5.3.0 is released

Closes #1575 

## Description

This PR is based on https://github.com/twbs/bootstrap/pull/38410 created upstream.

Depending on the Boosted version, the modifications are different:
- [x] 3.3: Banner was already here so changed "The first Boosted 4 beta is here!" to "There's a newer version of Boosted!"
- [x] 3.4: Banner was already here so changed "Looking for Boosted 4?" to "There's a newer version of Boosted!"
- [x] 4.0 → 4.6: No banner so added a `.d-block` to the `<header>` plus the following code to display the banner:

```html
 <nav role="navigation" class="navbar navbar-expand-md py-0" aria-label="Complementary navigation">
  <div class="container-fluid nav-item justify-content-center">
    <ul class="navbar-nav">
      <li class="nav-item border-0">
        <a href="https://boosted.orange.com" class="nav-link">There's a newer version of Boosted!</a>
      </li>
    </ul>
  </div>
</nav>
```

Note: From 4.4 to 4.6, the link is slightly different to have a correct visible focus rendering: `<a href="https://boosted.orange.com" class="nav-link pr-1 pr-sm-2">`

- [x] 5.0 → 5.1: No banner so added a new section after the header. Otherwise, the search bar would be hidden by an extra div on top of the header. Here's the new section:

```html
<nav role="navigation" class="navbar navbar-expand-md py-0 fixed-top navbar-dark pe-none" aria-label="Complementary navigation">
  <div class="container-fluid nav-item justify-content-center">
    <ul class="navbar-nav">
      <li class="nav-item border-0">
        <a href="https://boosted.orange.com" class="nav-link pe-auto"><span class="d-none d-sm-block fs-6">There's a newer version of Boosted!</span><span class="d-sm-none fs-6">Newer version of Boosted!</span></a>
      </li>
    </ul>
  </div>
</nav>
```

Please note that the sentence is reduced in mobile mode and that the skip links require and extra dark background in order to be on top of our new banner:

```html
<a class="d-inline-flex p-2 ms-3 bg-dark" href="#content">Skip to main content</a>
<!-- ... -->
<a class="d-none d-md-inline-flex p-2 bg-dark" href="#bd-docs-nav">
```

- [x] 5.2: No banner so added a new section after the header. Otherwise, the search bar would be hidden by an extra div on top of the header. Here's the new section:

```html
<nav role="navigation" class="navbar navbar-expand-md py-0 fixed-top navbar-dark pe-none shadow-none" aria-label="Complementary navigation">
  <div class="container-fluid nav-item justify-content-center">
    <ul class="navbar-nav">
      <li class="nav-item border-0">
        <a href="https://boosted.orange.com" class="nav-link pe-auto"><span class="d-none d-sm-block fs-6">There's a newer version of Boosted!</span><span class="d-sm-none fs-6">Newer version of Boosted!</span></a>
      </li>
    </ul>
  </div>
</nav>
```

⚠️ On top of that this PR removes `v4-test` directory which isn't used anymore. `v4-alpha*` directories will be removed later on because they need a modification in the versions dropdown menu in v3 documentations.

## Local testing

```
php -S localhost:8008 # or any other command to launch a local server
```

Then check all the modified Boosted versions docs:
- http://localhost:8008/docs/3.3/index.html
- http://localhost:8008/docs/3.4/index.html
- http://localhost:8008/docs/4.1/about/boosted-history/index.html
- http://localhost:8008/docs/4.2/about/boosted-history/index.html
- http://localhost:8008/docs/4.3/about/boosted-history/index.html
- http://localhost:8008/docs/4.4/about/boosted-history/index.html
- http://localhost:8008/docs/4.5/about/boosted-history/index.html
- http://localhost:8008/docs/4.6/about/boosted-history/index.html
- http://localhost:8008/docs/5.0/about/brand/index.html
- http://localhost:8008/docs/5.1/about/brand/index.html
- http://localhost:8008/docs/5.2/about/brand/index.html

### CODE REVIEW

File: v4-test/docs/4.0/about/boosted-history/index.html
Comment (julien-deramond) [inline v4-test/docs/4.0/about/boosted-history/index.html:1]: Maybe we can drop this `v4-test` directory entirely 🤷

**Review by louismaximepiton — COMMENTED** [review]
The focus order seems a bit weird sometimes but I can't think of another way so let's go this way.

Issue on 4.x and 5.x: There's a small border-top on `.nav-item` on mobile (more visible on 4.4-4.6)
Issue on 4.4+: The focus isn't pretty, is there a way to have `.p-2` on it ?
Remark on 5.2: The minimized version is particularly stacked

File: v4-test/docs/4.0/about/boosted-history/index.html
Comment (louismaximepiton) [reply v4-test/docs/4.0/about/boosted-history/index.html:1]: Never seen it before you mention it so I guess it'd be fine + Bootstrap doesn't have it.

**Comment by julien-deramond** [comment]
Latest modifications:
- [x] Removed `docs/**/examples/**` modifications (except `docs/**/examples/index.html`) that were issues
- [x] From 4.0 to 5.1, `.nav-item` needed a `.border-0`
- [x] From 4.4 to 4.6, link needed `.pr-1 .pr-sm-2` for a correct visible focus rendering
- [x] 5.2 stacked rendering is kept as-is because supra bar would need some extra modifications for the side navigation and it's too "complex" and far from the original design

**Review by louismaximepiton — APPROVED** [review]
LGTM 🚀

**Pull request merged**